File: app/src/main/java/com/fjsdkqwj/pfdioewjnsd/ui/WelcomeActivity.java
package com.fjsdkqwj.pfdioewjnsd.ui;

import android.content.DialogInterface;
import android.os.Bundle;
import android.os.Handler;
import android.text.TextUtils;
import android.view.KeyEvent;

import com.fjsdkqwj.pfdioewjnsd.R;
import com.fjsdkqwj.pfdioewjnsd.api.RetrofitManager;
import com.fjsdkqwj.pfdioewjnsd.base.BaseActivity;
import com.fjsdkqwj.pfdioewjnsd.util.SharePreferencesUtil;
import com.fjsdkqwj.pfdioewjnsd.util.StaticUtil;
import com.fjsdkqwj.pfdioewjnsd.util.StatusBarUtil;
import com.fjsdkqwj.pfdioewjnsd.util.WelcomeRemindDialog;

public class WelcomeActivity extends BaseActivity {

    private Bundle bundle;

    private boolean started = false, isResume = false;

    private String phone = "";

    private WelcomeRemindDialog welcomeRemindDialog;

    @Override
    protected void onResume() {
        isResume = true;
        super.onResume();
        new Handler().postDelayed(() -> isResume = false, 500);
    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_welcome;
    }

    @Override
    public void initListener() {

    }

    @Override
    public void initData() {
        StatusBarUtil.setTransparent(this, false);
        started = SharePreferencesUtil.getBool("started");
        phone = SharePreferencesUtil.getString("phone");
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                startPage();
            }
        }, 500);
    }

    private void showDialog() {
        welcomeRemindDialog = new WelcomeRemindDialog(this);
        welcomeRemindDialog.setOnKeyListener(new DialogInterface.OnKeyListener() {
            @Override
            public boolean onKey(DialogInterface dialog, int keyCode, KeyEvent event) {
                if (keyCode == KeyEvent.KEYCODE_BACK && !isResume) {
                    finish();
                    return false;
                }
                return true;
            }
        });
        welcomeRemindDialog.setOnListener(new WelcomeRemindDialog.OnListener() {

            @Override
            public void topBtnClicked() {
                SharePreferencesUtil.saveBool("started", true);
                welcomeRemindDialog.dismiss();
                StaticUtil.startActivity(WelcomeActivity.this, LoginActivity.class, null);
                finish();
            }

            @Override
            public void bottomBtnClicked() {
                finish();
            }

            @Override
            public void clickedZcxy() {
                bundle = new Bundle();
                bundle.putInt("tag", 1);
                bundle.putString("url", RetrofitManager.ZCXY);
                StaticUtil.startActivity(WelcomeActivity.this, UserYsxyActivity.class, bundle);
            }

            @Override
            public void clickedYsxy() {
                bundle = new Bundle();
                bundle.putInt("tag", 2);
                bundle.putString("url", RetrofitManager.YSXY);
                StaticUtil.startActivity(WelcomeActivity.this, UserYsxyActivity.class, bundle);
            }
        });
        welcomeRemindDialog.show();
    }

    private void startPage() {
        if (started) {
            if (!TextUtils.isEmpty(phone)) {
                StaticUtil.startActivity(WelcomeActivity.this, MainActivity.class, null);
            } else {
                StaticUtil.startActivity(WelcomeActivity.this, LoginActivity.class, null);
            }
            finish();
        } else {
            showDialog();
        }
    }
}

File: app/src/main/java/com/asvsdfer/new_master_code_kotlin/ui/activity/WelcomeActivity.kt
package com.asvsdfer.new_master_code_kotlin.ui.activity

import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.os.Handler
import android.text.TextUtils
import android.view.KeyEvent
import androidx.appcompat.app.AppCompatActivity
import com.asvsdfer.new_master_code_kotlin.R
import com.asvsdfer.new_master_code_kotlin.logic.network.ServiceCreator
import com.asvsdfer.new_master_code_kotlin.ui.view.WelcomePageDialog
import com.asvsdfer.new_master_code_kotlin.util.PreferencesOpenUtil
import com.asvsdfer.new_master_code_kotlin.util.StatusBarUtil
import com.asvsdfer.new_master_code_kotlin.util.jumpActivity

class WelcomeActivity : AppCompatActivity(){

    private var started: Boolean = false

    private  var isResume = false

    private var phone = ""

    private var welcomePageDialog: WelcomePageDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        StatusBarUtil.setTransparent(this, false)
        setContentView(R.layout.activity_welcome)
        started = PreferencesOpenUtil.getBool("started")!!
        phone = PreferencesOpenUtil.getString("phone").toString()
        Handler().postDelayed({ startPage() }, 500)
    }

    override fun onResume() {
        isResume = true
        super.onResume()
        Handler().postDelayed({ isResume = false }, 500)
    }

    private fun showDialog() {
        welcomePageDialog = WelcomePageDialog(this)
        welcomePageDialog?.setOnKeyListener(DialogInterface.OnKeyListener { dialog, keyCode, event ->
            if (keyCode == KeyEvent.KEYCODE_BACK && !isResume) {
                finish()
                return@OnKeyListener false
            }
            true
        })
        welcomePageDialog?.setOnListener(object : WelcomePageDialog.OnListener {
            override fun topBtnClicked() {
                PreferencesOpenUtil.saveBool("started", true)
                welcomePageDialog?.dismiss()
                jumpActivity<LoginActivity>(this@WelcomeActivity)
                finish()
            }

            override fun bottomBtnClicked() {
                finish()
            }

            override fun clickedZcxy() {
                jumpActivity<WebViewActivity>(this@WelcomeActivity){
                    putExtra("tag", 1)
                    putExtra("url", ServiceCreator.ZCXY)
                }
            }

            override fun clickedYsxy() {
                jumpActivity<WebViewActivity>(this@WelcomeActivity){
                    putExtra("tag", 2)
                    putExtra("url", ServiceCreator.YSXY)
                }
            }
        })
        welcomePageDialog?.show()
    }

    private fun startPage() {
        if (started) {
            if (!TextUtils.isEmpty(phone)) {
                jumpActivity<MainActivity>(this@WelcomeActivity)
            } else {
                jumpActivity<LoginActivity>(this@WelcomeActivity)
            }
            finish()
        } else {
            showDialog()
        }
    }

}